blocks/math/include/gnuradio-4.0/math/Math.hpp

```diff
@@ -115,6 +125,221 @@ using Multiply = MathOpMultiPortImpl<T, std::multiplies<T>>;
+
+    Depending on the operator op this block computes:
+    - Negate: out = - in
+    - Not: out = ~ in
+    - Abs: out = abs(in)
+    )"">;
+
+    // ports
+    PortIn<T>  in;
+    PortOut<T> out;
+
+    GR_MAKE_REFLECTABLE(MathOpSinglePortImpl, in, out);
+
+    [[nodiscard]] constexpr T processOne(const auto& a) const noexcept { return op{}(a); }
+};
+
+template<typename T>
+using Negate = MathOpSinglePortImpl<T, std::negate<T>>;
+template<typename T>
+using Not = MathOpSinglePortImpl<T, std::bit_not<T>>;
+
+template<class T = void>
+struct abs_op {
+    constexpr T operator()(const T& v) const {
+        if constexpr (std::is_unsigned_v<T>) {
+            // |v| == v  ➜ avoid ambiguous std::abs and silence -Wabsolute-value
+            return v;
+        } else if constexpr (std::is_integral_v<T>) {
+            // std::abs promotes to int/long long; cast back to T
+            using Promoted = std::conditional_t<(sizeof(long long) > sizeof(int) &&
+                                                 sizeof(T) > sizeof(int)),
+                                                long long,
+                                                int>;
+            return static_cast<T>(std::abs(static_cast<Promoted>(v)));
+        } else {
+            // float, double, long double: return type is already correct
+            return std::abs(v);
+        }
+    }
+};
+
+template<typename T>
+using Abs = MathOpSinglePortImpl<T, abs_op<T>>;
+
+/* ------------------------------------------------------------------ *
+ *  Integrate : running sum over N samples, output 1 value, decimate   *
+ * ------------------------------------------------------------------ */
+
+ GR_REGISTER_BLOCK("gr::blocks::math::Integrate",
+                  gr::blocks::math::Integrate,
+                  ([T]),
+                  [ uint8_t, uint16_t, uint32_t, uint64_t,
+                    int8_t,  int16_t,  int32_t,  int64_t,
+                    float, double ])
+
+template<typename T>
+requires std::is_arithmetic_v<T>
+struct Integrate : gr::Block<Integrate<T>>
+{
+    /* ---------- Doc string ---------------------------------------- */
+    using Description = Doc<R""(
+@brief Integrate successive samples and decimate.
+
+For every *decim* input items the block outputs the sum of those
+*decim* items and then resets the accumulator.
+
+Example, decim = 4  
+in : 1 2 3 4 5 6 7 8   →  out : 10 26
+)"">;
+
+    /* ---------- Ports & settings ---------------------------------- */
+    PortIn<T>  in;
+    PortOut<T> out;
+
+    Annotated<gr::Size_t,
+              "decim",
+              Visible,
+              Doc<"decimation / integration length">,
+              Limits<1UZ, (1UZ << 20)> >    // ← space before last '>'
+        decim = 1UZ;
+
+    GR_MAKE_REFLECTABLE(Integrate, in, out, decim);
+
+    /* ---------- State --------------------------------------------- */
+    T           _acc  = T(0);
+    std::size_t _seen = 0;
```

Comment:
I'm not sure what the `_seen` variable is doing,  Looking at the code it looks like it is always 0.  Could you explain it?